src/network/GrpcEndpoint.ts

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Endpoint } from "./AbstractEndpoint";
+import { URIParseConfig } from "./Network.util";
+import { URL, URLSearchParams } from "url";
+
+export class GrpcEndpoint extends Endpoint {
+  constructor(url: string) {
+    super(url);
+    this._authority = URIParseConfig.DEFAULT_AUTHORITY;
+
+    this._parsedUrl = new URL(this.preprocessUri(url));
+    this.validatePathAndQuery();
+
```

Comment:
We can convert these to const, instead of class variables, and pass them as parameters to the different functions.

E.g.,
```
this._scheme=GrpcEndpoint.getScheme(parsedUri)
this._port = GrpcEndpoint.getPort(parsedUri, scheme);
```

Reply:
It should be fine, thanks for checking!